utils/swift_build_support/swift_build_support/products/benchmarks.py

```diff
@@ -100,11 +100,17 @@ def get_dependencies(cls):
 
 
 def _get_toolchain_path(host_target, product, args):
-    # TODO check if we should prefer using product.install_toolchain_path
     # this logic initially was inside run_build_script_helper
     # and was factored out so it can be used in testing as well
 
-    toolchain_path = product.host_install_destdir(host_target)
+    install_destdir = args.install_destdir
+    if swiftpm.SwiftPM.has_cross_compile_hosts(args):
+        install_destdir = swiftpm.SwiftPM.get_install_destdir(args,
+                                                              host_target,
+                                                              product.build_dir)
+    toolchain_path = targets.toolchain_path(install_destdir,
+                                            args.install_prefix)
```

Comment:
Note the prefix added for Darwin in the next couple lines, that's going to break this usual formulation there, [which is why I didn't mess with it last year](https://github.com/apple/swift/pull/36917/files#r688940409).